### ISSUE TITLE
New SPIR SPV_INTEL_vector_compute extension

### ISSUE THREAD

**Issue by LifeIsStrange** [servo/webrender#4030]
https://www.phoronix.com/scan.php?page=news_item&px=Intel-IGC-Vector-Backend

https://github.com/intel/llvm/pull/1612/files

This could revolutionize 2D rendering performance (and/or efficiency) on Intel igpus (most of the market) so you should keep an eye on it :}

**Comment by kvark** [comment]
Thank you for the link! Do you have an idea on how this would be integrated into WebRender?
It seems to me that `SPV_INTEL_vector_compute` is an OpenCL-focused SPIR-V extension. WebRender currently uses neither OpenCL or SPIR-V.

**Comment by LifeIsStrange** [comment]
@kvark good catch! Well this extension might be SPIR-V so theoretically API agnostic (e.g Vulkan supports SPIR-V and you have or will have support for Vulkan)
But it might be an openCL/syCL only extension (unclear), if so yeah webrender would to to add support for such API when supported. There might be other potential performance improvements coming from openCL use in general and its support is quite widespread in 2020.
If I recall correctly, pathfinder did use openCL at some point (I believe that's no longer the case)

In the far long term Vulkan and open/SYCL should merge https://pcper.com/2017/05/follow-up-neil-trevett-and-tom-olson-from-khronos-group-discuss-opencl-and-vulkan-roadmap/

*Do you have an idea on how this would be integrated into WebRender?* No idea :)
